sim/core/cast.go

```diff
@@ -146,27 +146,28 @@ func (cast *Cast) init(sim *Simulation) {
 // Start casting the spell. Return value indicates whether the spell successfully
 // started casting.
 func (cast *Cast) startCasting(sim *Simulation, onCastComplete OnCastComplete) bool {
-
-	switch cast.Cost.Type {
-	case stats.Mana:
-		if cast.Character.CurrentMana() < cast.Cost.Value {
-			if sim.Log != nil {
-				cast.Character.Log(sim, "Failed casting %s, not enough mana. (Current Mana = %0.03f, Mana Cost = %0.03f)",
-					cast.ActionID, cast.Character.CurrentMana(), cast.Cost.Value)
+	if cast.Cost.Value > 0 {
```

Comment:
Adding the condition here is fine, but make it >= 0. We still want to call the SpendRage/SpendEnergy/etc at 0 so that it gets logged, especially because the logs are used to create the timeline graphs in detailed results.

Reply:
In that case just remove the extra condition, it's not saving anything meaningful, anyhow. I have checked "Allow edits by maintainers", so this should work (easier then ping-ponging)

Reply:
Made the edits